app/models/concerns/media_facebook_engagement_metrics.rb

```diff
@@ -9,14 +9,28 @@ module MediaFacebookEngagementMetrics
 
   def get_metrics_from_facebook
     facebook_id = self.data['uuid'] if is_a_facebook_post?
-    self.class.get_metrics_from_facebook(self.original_url, ApiKey.current&.id, 0, facebook_id)
+    MetricsWorker.perform_async(self.original_url, ApiKey.current&.id, 0, facebook_id)
   end
 
   def is_a_facebook_post?
     self.data && self.data['provider'] == 'facebook' && self.data['type'] == 'item'
   end
 
   module ClassMethods
+    def get_metrics_from_facebook(url, key_id, count = 0, facebook_id = nil)
```

Comment:
Just moving this method up so that it reads a little clearer, since this method is the main entrypoint